Add the IISMiddleware via a IStartupFilter.

Tratcher opened the pull request:
#95 @DamianEdwards @davidfowl 

I'll submit a separate PR to Hosting to make ConfigureServices callable multiple times.

davidfowl commented on src/Microsoft.AspNetCore.Server.IISIntegration/IISAddressExtensions.cs:
```diff
@@ -16,7 +18,7 @@ public static class IISAddressExtensions
         /// <param name="app"></param>
         /// <returns></returns>
-        public static IWebHostBuilder UseIISUrl(this IWebHostBuilder app)
+        public static IWebHostBuilder UseIIS(this IWebHostBuilder app)
```
pass `Func<string, string>` and use that in the tests.

Tratcher commented:
Updated. Environment variables have been abstracted. I've also removed the IApplicationBuilder extensions, we only use the IWebHostBuilder extension now.

davidfowl commented on test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs:
```diff
@@ -75,18 +106,19 @@ public class IISMiddlewareTests
-                        return next();
-                    });
-                    app.UseIIS(new IISOptions
+                    services.AddSingleton(new IISOptions()
```
What? Shouldn't this be configure options?

Tratcher replied:
Oh, is that how that's supposed to work? Updated.

davidfowl commented on src/Microsoft.AspNetCore.Server.IISIntegration/IISWebHostExtensions.cs:
```diff
@@ -0,0 +1,47 @@
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            var port = app.GetSetting(ServerPort);
```
Nice and testable

davidfowl commented on test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs:
```diff
@@ -33,25 +37,50 @@ public class IISMiddlewareTests
+
+            var builder = new WebHostBuilder()
+                .UseSetting("TOKEN", "TestToken")
+                .UseSetting("PORT", "12345")
```
Nice

davidfowl commented:
:shipit:
